src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -284,6 +308,7 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
     authorization = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
     headers = authorization
     headers['content-type'] = 'application/octet-stream'
+    headers['User-Agent'] = 'azure-cli:{}'.format(__version__)
```

Comment:
For setting agent consistently, please check out https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L499